packages/grafana-ui/src/components/ConfirmModal/ConfirmContent.tsx

```diff
@@ -0,0 +1,140 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useRef, useState } from 'react';
+import { useForm } from 'react-hook-form';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+
+import { useStyles2 } from '../../themes';
+import { Button, ButtonVariant } from '../Button';
+import { Input } from '../Input/Input';
+import { Box } from '../Layout/Box/Box';
+import { Stack } from '../Layout/Stack/Stack';
+import { JustifyContent } from '../Layout/types';
+import { ResponsiveProp } from '../Layout/utils/responsiveness';
+
+export interface ConfirmContentProps {
+  /** Modal content */
+  body: React.ReactNode;
+  /** Modal description */
+  description?: React.ReactNode;
+  /** Text for confirm button */
+  confirmText: string;
+  /** Text for dismiss button */
+  dismissText?: string;
+  /** Variant for dismiss button */
+  dismissVariant?: ButtonVariant;
+  /** Text user needs to fill in before confirming */
+  fillToConfirmText?: string;
+  /** Text for alternative button */
+  alternativeText?: string;
```

Comment:
Following Josh's comment, we could rename all the button props to be clearer (and maybe move them together)
`alternativeText` -> `alternativeButtonLabel`
`dismissVariant` -> `dismissButtonVariant`